src/java/org/apache/cassandra/index/SecondaryIndexManager.java

```diff
@@ -316,7 +317,13 @@ public void checkQueryability(Index.QueryPlan queryPlan)
         for (Index index : queryPlan.getIndexes())
         {
             if (!isIndexQueryable(index))
+            {
+                // In Astra index can be queryable during index build, thus we need to check both not queryable and building
+                if (isIndexBuilding(index))
```

Comment:
Hmmm, we probably need to use instead directly the full rebuild status here instead? Same as in `filterForQuery`

Reply:
You're right. I don't think we'd want to count the cases where we aren't doing a full/initial build.